### PR TITLE
Delete emarkdown

### PR DESCRIPTION
**Pull request by vkatsuba**
* Replace `emarkdown` to simple `RegEx` cases for support Windows 
* Adapt trim new lines for Windows
* Remove `provider_hooks` for Windows
* Added Windows to CI

### CODE REVIEW

File: src/adapter/markdown_adapter.erl [hunk 1]
```diff
@@ -31,5 +31,12 @@
...
-    html_adapter:adapt(Line1).
+    %% Prepare links
+    L0 = binary:replace(Line,
+                        [<<"`">>, <<"[">>, <<"]">>, <<"(">>, <<")">>],
+                        <<" ">>,
+                        [global]),
+    %% Delete links in line
+    L1 = re:replace(L0, "(http\S*?)(\.*?\s)", " ", [global, {return, binary}]),
+    %% Delete other symbols
+    binary:replace(L1, [<<"`">>, <<"<">>, <<">">>, <<"*">>, <<"#">>], <<>>, [global]).
```
Comment (elbrujohalcon) [inline src/adapter/markdown_adapter.erl:42]: What about `~`. Should we account for that one as well?

**Review by elbrujohalcon — APPROVED** [review]
I approve, in general, but I think we'll eventually find some edge cases that we can't handle with this change.
But, that trade off is worth it when thinking about adding this to projects like erlang-ls.

File: src/adapter/markdown_adapter.erl (hunk 1 above)
Comment (vkatsuba) [reply src/adapter/markdown_adapter.erl:42]: well, I suppose in future we will add more changes here because I don't see trim of `~` and some other chars in `sheldon`.

Comment (vkatsuba) [reply src/adapter/markdown_adapter.erl:42]: Current PR is focused on fixing issue of Windows compile + CT passing without code changes in test cases - I suppose this is a good start.

**Comment by vkatsuba** [comment]
> I approve, in general, but I think we'll eventually find some edge cases that we can't handle with this change.
> But, that trade off is worth it when thinking about adding this to projects like erlang-ls.

Yes I suppose with markdown as with HTML adapters we will have some issues. But they can be fixed when we will find those problems. For now I suppose it make sense to keep it as is.

File: src/adapter/markdown_adapter.erl (hunk 1 above)
Comment (vkatsuba) [reply src/adapter/markdown_adapter.erl:42]: BTW:
```erlang
1> sheldon:check("~").
#{bazinga =>
      <<"The only way you'd be able to make a contribution to science is if they resume sending chimps into space.">>,
  misspelled_words =>
      [#{candidates =>
             ["2","A","a","B","b","C","c","D","d","E","e","F","f","G",
              "g","H","h","I","i","J","j","K","k","L","l",
              [...]|...],
         line_number => 1,word => "~"}]}
```

**Comment by ferigis** [comment]
same concern as Brujo mentioned but approved :) next version bump I would suggest to increase the minor at least.

**Review by ferigis — APPROVED** [review]

**Comment by vkatsuba** [comment]
> same concern as Brujo mentioned but approved :) next version bump I would suggest to increase the minor at least.

Yep, will be done a lot of changes and improvements 🙃
